src/PluginStartupRegistration.cpp

```diff
@@ -14,12 +14,118 @@
 
 #include <wx/log.h>
 #include <wx/app.h>
+#include <wx/stattext.h>
+#include <wx/gauge.h>
+#include <wx/button.h>
+#include <wx/timer.h>
+#include <wx/sizer.h>
 
 #include "PluginManager.h"
 #include "PluginDescriptor.h"
-#include "widgets/ProgressDialog.h"
-#include "widgets/wxWidgetsWindowPlacement.h"
+#include "widgets/wxPanelWrapper.h"
 
+namespace
+{
+   enum {
+      OnPluginScanTimeout = wxID_HIGHEST + 1,
+   };
+
+   class PluginScanDialog : public wxDialogWrapper
+   {
+      wxStaticText* mText{nullptr};
+      wxStaticText* mElapsed{nullptr};
+      wxGauge* mProgress{nullptr};
+      std::chrono::system_clock::time_point mStartTime;
+   public:
+      
+      PluginScanDialog(
+         wxWindow* parent,
+         wxWindowID winid,
+         const TranslatableString& title)
+         : wxDialogWrapper(parent, winid, title)
+      {
+         auto rootSizer = std::make_unique<wxBoxSizer>(wxVERTICAL);
```

Comment:
If the schedule were more relaxed, I would ask you to learn the use of ShuttleGui rather than making so many direct calls to wxSizer methods.  But the task of rewriting this will wait until another release and perhaps be someone else's problem.